test/box-tap/gh-6634-different-log-on-tuple-new-and-free.test.lua

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env tarantool
+
+local tap = require('tap')
+local test = tap.test('gh-6634-different-logs')
+local test_run = require('test_run').new()
+
+test:plan(3)
+-- create tarantool instance
+test:is(
+        test_run:cmd("create server second with script='box/lua/cfg_debug_log.lua'\n"),
+        true, 'instance created'
+)
+test:is(
+        test_run:cmd('start server second'),
+        true, 'instance started'
+)
+
+test_run:eval('second', "s = box.schema.create_space('test')")
+test_run:eval('second', "s:create_index('pk')")
+test_run:eval('second', 's:replace{0, 0, 0, 0, 0, 0, 0}')
+local str = test_run:grep_log('second', 'tuple_new_raw_impl%w*%(%d+%) = 0x%x+$', 1024)
+local new_tuple_address = string.match(str, '0x%x+$')
+new_tuple_address = string.sub(new_tuple_address, 3)
+
+test_run:eval('second', "collectgarbage('collect')")
+
+-- Replace previously inserted tuple with new one
+-- to activate memtx_gc and delete the tuple.
+test_run:eval('second', '_ = s:replace{0, 1}')
+
+test_run:eval('second', "collectgarbage('collect')")
+
+-- Check if last deleted tuple is the tuple inserted above.
+str = test_run:grep_log('second', 'tuple_delete%w*%(0x%x+%)')
+local deleted_tuple_address = string.match(str, '0x%x+%)')
+deleted_tuple_address = string.sub(deleted_tuple_address, 3, -2)
+test:ok(new_tuple_address == deleted_tuple_address)
+
+test_run:cmd('stop server second with cleanup=1')
+
+os.exit(test:check() and 0 or 1)
```

Comment:
Please rewrite the test in luatest. TAP tests are not permitted anymore.

Reply:
done